fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/predicates/PredicateWithValueAndRanges.java

```diff
@@ -331,7 +344,8 @@ public Optional<PredicateMapping> impliesCandidatePredicate(@NonNull final Alias
         if (semanticEquals(candidatePredicate, aliasMap)) {
```

Comment:
Shouldn't this be a `semanticEquals()` on a predicate that is _translated_? Or is the `this` already translated?

Reply:
I think the semantic equality is correctly (as it is using the proper alias mapping), it is checking whether the `this` (query) predicate is semantically equal to the candidate predicate it proceeds to return a mapping. I think this had been always the only check needed for establishing a predicate mapping for non-trivial predicates (tautology/contradiction) before the introduction of `PredicateWithValuesAndRanges`.

Nevertheless, I think the translation here is incorrect, because we will never apply the query predicate as a residual since it is semantically equal to the candidate predicate. In other words, the mapping should not contain a translated predicate here if I am not mistaken.